qiskit/ignis/verification/topological_codes/fitters.py

```diff
@@ -60,106 +63,120 @@ def __init__(self, code, S=None):
 
         if S:
             self.S = S
+        elif not brute and "_get_all_processed_results" in dir(code):
+            self.S = self._make_syndrome_graph(
+                results=code._get_all_processed_results()
+            )
         else:
             self.S = self._make_syndrome_graph()
 
     def _separate_string(self, string):
 
         separated_string = []
-        for syndrome_type_string in string.split('  '):
-            separated_string.append(syndrome_type_string.split(' '))
+        for syndrome_type_string in string.split("  "):
+            separated_string.append(syndrome_type_string.split(" "))
         return separated_string
 
-    def _string2nodes(self, string, logical='0'):
+    def _string2nodes(self, string, logical="0"):
 
         separated_string = self._separate_string(string)
         nodes = []
         for syn_type, _ in enumerate(separated_string):
-            for syn_round in range(
-                    len(separated_string[syn_type])):
-                elements = \
-                    separated_string[syn_type][syn_round]
+            for syn_round in range(len(separated_string[syn_type])):
+                elements = separated_string[syn_type][syn_round]
                 for elem_num, element in enumerate(elements):
-                    if (syn_type == 0 and element != logical) or (syn_type != 0 and element == '1'):
-                        nodes.append((syn_type,
-                                      syn_round,
-                                      elem_num))
+                    if (syn_type == 0 and element != logical) or (
+                        syn_type != 0 and element == "1"
+                    ):
+                        nodes.append((syn_type, syn_round, elem_num))
         return nodes
 
-    def _make_syndrome_graph(self):
-        """
-        This method injects all possible Pauli errors into the circuit for
-        ``code``.
-
-        This is done by examining the qubits used in each gate of the
-        circuit for a stored logical 0. A graph is then created with a node
-        for each non-trivial syndrome element, and an edge between all such
-        elements that can be created by the same error.
-        """
+    def _make_syndrome_graph(self, results=None):
 
         S = rx.PyGraph(multigraph=False)
+        if results:
+
+            node_map = {}
+            for string in results:
+                nodes = self._string2nodes(string)
+                for node in nodes:
+                    if node not in node_map:
+                        node_map[node] = S.add_node(node)
```

Comment:
This might be slightly more efficient, but not a big deal either way

```suggestion
                nodes = self._string2nodes(string)
                node_ids = S.add_nodes_from(nodes)
                for node, node_id in zip(nodes, node_ids):
                    if node not in node_map:
                        node_map[node] = node_id
```